widget/embedded/src/pages/ConfirmSwapPage.tsx

```diff
@@ -224,19 +231,41 @@ export function ConfirmSwapPage(props: PropTypes) {
         )
       );
 
-    const routeWalletsChanged = confirmSwapDisabled(
-      fetchingConfirmationRoute,
-      !!customDestination,
-      customDestination,
-      bestRoute,
-      selectedWallets,
-      destination
+    let routeWalletsChanged = false;
+
+    let requiredWallets = getRequiredWallets(bestRoute);
+
+    const lastStepToBlockchain =
+      bestRoute?.result?.swaps[bestRoute.result.swaps.length - 1].to.blockchain;
+
+    const isLastWalletRequired = !!bestRoute?.result?.swaps.find(
+      (swap) => swap.from.blockchain === lastStepToBlockchain
     );
 
+    if (!isLastWalletRequired) {
+      requiredWallets = requiredWallets.slice(
+        -requiredWallets.length,
+        requiredWallets.length - 1
+      );
+    }
+
+    const allRequiredWalletsSelected = requiredWallets.every((requiredWallet) =>
+      selectedWallets.find(
+        (selectedWallet) => selectedWallet.chain === requiredWallet
+      )
+    );
+
+    if (!allRequiredWalletsSelected) {
+      routeWalletsChanged = true;
+    }
+
     if (selectedWalletDisconnected || routeWalletsChanged) {
       setShowWallets(true);
+      setTimeout(() => {
+        setRouteWalletConfirmed(false);
```

Comment:
It seems when I approve confirm wallet selection modal, it will be redirected to home page by mistake.